pycbc/types/timeseries.py

```diff
@@ -595,15 +595,31 @@ def gate(self, time, window=0.25, method='taper', copy=True,
             # Uses the hole-filling method of
             # https://arxiv.org/pdf/1908.05644.pdf
             from pycbc.strain.gate import gate_and_paint
+            from pycbc.waveform.utils import apply_fd_time_shift
             if invpsd is None:
                 # These are some bare minimum settings, normally you
                 # should probably provide a psd
                 invpsd = 1. / self.filter_psd(self.duration/32, self.delta_f, 0)
             lindex = int((time - window - self.start_time) / self.delta_t)
-            rindex = lindex + int(2 * window / self.delta_t)
+            rindex = int((time + window - self.start_time) / self.delta_t)
             lindex = lindex if lindex >= 0 else 0
             rindex = rindex if rindex <= len(self) else len(self)
-            return gate_and_paint(data, lindex, rindex, invpsd, copy=False)
+            # shift data such that gate_end_time lands on a specific data sample
+            fdata = data.to_frequencyseries()
+            rindex_time = data.sample_times[rindex]
+            offset = rindex_time - (time + window)
```

Comment:
As we discussed on the call, calculate offset at line 607. Then only do the time shifting if offset != 0. Also make `copy=False` at all points here, since the copy was already done at the beginning of the function.